src/ctapipe/utils/astro.py

```diff
@@ -4,63 +4,239 @@
-    data downloads.
+    Each catalog entry is represented as a namedtuple `CatalogInfo` containing:
+
+    Attributes
+    ----------
+    directory : str
+        The directory path of the catalog in the Vizier database.
+    coordinates : dict
+        A dictionary containing the coordinate frame, epoch, and column names for RA and DE.
+    columns : list of str
+        A list of columns to be retrieved from the catalog.
+    record : str
+        A name of the catalog file in the cache.
+    """
 
-    The included Yale bright star catalog contains all 9096 stars, excluding the
-    Nova objects present in the original catalog and is complete down to magnitude
-    ~6.5, while the faintest included star has mag=7.96. :cite:p:`bright-star-catalog`
+    Yale = CatalogInfo(
+        directory="V/50/catalog",
+        coordinates={
+            "frame": "icrs",
+            "epoch": "J2000.0",
+            "RA": {"column": "RAJ2000", "unit": "hourangle"},
+            "DE": {"column": "DEJ2000", "unit": "deg"},
+        },
+        #: Vmag is mandatory (used for initial magnitude cut)
+        columns=["RAJ2000", "DEJ2000", "pmRA", "pmDE", "Vmag", "HR"],
+        record="yale_bright_star_catalog",
+    )  #: Yale bright star catalogue
+    Hipparcos = CatalogInfo(
+        directory="I/239/hip_main",
+        coordinates={
+            "frame": "icrs",
+            "epoch": "J1991.25",
+            "RA": {"column": "RAICRS", "unit": "deg"},
+            "DE": {"column": "DEICRS", "unit": "deg"},
+        },
+        #: Vmag is mandatory (used for initial magnitude cut)
+        columns=["RAICRS", "DEICRS", "pmRA", "pmDE", "Vmag", "BTmag", "HIP"],
+        record="hipparcos_star_catalog",
+    )  #: HIPPARCOS catalogue
+
+
+def select_stars(
+    stars: Table,
+    pointing: SkyCoord = None,
+    radius: Quantity = None,
+    magnitude_cut: float = None,
+    band: str = "Vmag",
+) -> Table:
+    """
+    Utility function to filter stars based on magnitude and/or location.
 
     Parameters
     ----------
-    pointing: astropy Skycoord
-       pointing direction in the sky (if none is given, full sky is returned)
-    radius: astropy angular units
-       Radius of the sky region around pointing position. Default: full sky
-    magnitude_cut: float
-        Return only stars above a given magnitude. Default: None (all entries)
+    stars : astropy.table.Table
+        Table of stars, including magnitude and coordinates.
+    pointing : astropy.coordinates.SkyCoord, optional
+        Pointing direction in the sky. If None is given, the full sky is returned.
+    radius : astropy.units.Quantity, optional
+        Radius of the sky region around the pointing position. Default is the full sky.
+    magnitude_cut : float, optional
+        Return only stars above a given apparent magnitude. Default is None (all entries).
+    band : str, optional
+        Wavelength band to use for the magnitude cut. Options are 'Vmag' and 'B'. Default is 'Vmag'.
```

Comment:
Really `'B'`? Not `"BTmag"`?

Reply:
This should be changed. Default is Vmag as it is in all or almost all catalogs, but the B-band can be different (just B, Bmag, BTmag...) I will update this.